### ISSUE TITLE
Define SOS variables without a weight (ordering) vector

### ISSUE THREAD

**Issue by JipKim** [JuliaOpt/JuMP.jl#1916]
Making this feature request following the discussion in [https://discourse.julialang.org/t/sos1-variables-are-not-supported/22012/5](https://discourse.julialang.org/t/sos1-variables-are-not-supported/22012/5)

Can it be allowed to define SOS variables without a weight (ordering) vector? So that below two constraints (con1 & con2) represent the same thing?
```julia
@variable(m, x[1:3])
@constraint(m, con1, x in MOI.SOS1([1.0, 2.0, 3.0]))
@constraint(m, con2, x in MOI.SOS1())
```
In the case of SOS1 variables, ordering is less critical than SOS2 and an automatic ordering would be more convenient.

Thanks!

**Comment by odow** [comment]
@JipKim, this is an easy issue if you would like to make a contribution to JuMP.

You can follow the example of 
https://github.com/JuliaOpt/JuMP.jl/blob/c01802926c34aa30e5d272ac1063f1595084aa4a/src/sets.jl#L32-L33
and implement `JuMP.SOS1` and `JuMP.SOS2`.

Basically, something like
```julia
struct SOS1 <: AbstractVectorSet 
    weights;:Vector{Float64}
    function SOS1(weights::AbstractVector = Float64[])
        new(convert(Vector{Float64}, weights))
    end
end
function moi_set(set::SOS1, dim::Int)
    if length(set.weights) == 0
        return MOI.SOS1(collect(1:1.0:dim))
    elseif length(weights) == dim
        return MOI.SOS1(weights)
    else
        error("Weight vector in SOS1 is not of length($dim)")
    end
end
```

Do the same for `SOS2`, and then add tests like
https://github.com/JuliaOpt/JuMP.jl/blob/c01802926c34aa30e5d272ac1063f1595084aa4a/test/macros.jl#L250-L257

You can find more information about contributing here: https://github.com/JuliaOpt/JuMP.jl/blob/master/CONTRIBUTING.md

**Comment by odow** [comment]
Closed by #2212
